mirrord/layer-lib/src/process/windows/execution/pipes.rs

```diff
@@ -0,0 +1,289 @@
+                    format!("Failed to set inheritance for {} read handle", name),
+                )));
+            }
+
+            Ok(Self {
+                read_handle,
+                write_handle,
+                forwarding_thread: None,
+            })
+        }
+    }
+
+    /// Get write handle for child process
+    pub fn write_handle(&self) -> HANDLE {
+        self.write_handle
+    }
+
+    /// Get read handle for parent process  
+    pub fn read_handle(&self) -> HANDLE {
+        self.read_handle
+    }
+
+    /// Start background forwarding from read handle to destination
+    pub fn start_forwarding(&mut self, destination: HANDLE) -> LayerResult<()> {
+        if self.forwarding_thread.is_some() {
+            return Err(LayerError::IO(std::io::Error::new(
+                std::io::ErrorKind::AlreadyExists,
+                "Forwarding already started".to_string(),
+            )));
+        }
+
+        let src_handle = self.read_handle as usize;
+        let dst_handle = destination as usize;
+
+        let handle = thread::spawn(move || {
+            let mut buffer = vec![0u8; 8192];
+            let mut bytes_read = 0u32;
+            let mut bytes_written = 0u32;
+
+            loop {
+                unsafe {
+                    let read_success = ReadFile(
+                        src_handle as HANDLE,
+                        buffer.as_mut_ptr() as *mut _,
+                        buffer.len() as u32,
+                        &mut bytes_read,
+                        ptr::null_mut(),
+                    );
+
+                    // Check for actual read failure
+                    if read_success == 0 {
+                        let error = GetLastError();
+                        match error {
+                            ERROR_BROKEN_PIPE => {
+                                tracing::debug!(
+                                    "📡 Pipe closed (broken pipe), forwarding thread exiting"
+                                );
+                                break;
+                            }
+                            ERROR_NO_DATA => {
+                                tracing::debug!("📡 No data available, continuing...");
+                                std::thread::sleep(std::time::Duration::from_millis(10));
+                                continue;
+                            }
+                            _ => {
+                                tracing::debug!(
+                                    "📡 ReadFile failed with error {}, forwarding thread exiting",
+                                    error
+                                );
+                                break;
+                            }
+                        }
+                    }
+
+                    // Handle case where ReadFile succeeded but no bytes were read
+                    if bytes_read == 0 {
+                        tracing::debug!("📡 ReadFile succeeded but 0 bytes read, continuing...");
```

Comment:
let's please remove the emojis, they're super inconsistent, and not every font/terminal emulator supports this charset